server/src/main/java/com/linecorp/centraldogma/server/CentralDogmaConfig.java

```diff
@@ -75,6 +81,44 @@
  */
 public final class CentralDogmaConfig {
 
+    private static final Logger logger = LoggerFactory.getLogger(CentralDogmaConfig.class);
+
+    private static final List<ConfigValueConverter> CONFIG_VALUE_CONVERTERS;
+
+    static {
+        final ArrayList<ConfigValueConverter> configValueConverters = new ArrayList<>();
+        Streams.stream(ServiceLoader.load(ConfigValueConverter.class)).forEach(configValueConverters::add);
+        configValueConverters.add(DefaultConfigValueConverter.INSTANCE);
+        CONFIG_VALUE_CONVERTERS = ImmutableList.copyOf(configValueConverters);
+        logger.debug("Available {}s: {}", ConfigValueConverter.class.getName(), CONFIG_VALUE_CONVERTERS);
+    }
+
+    @Nullable
+    static String convertValue(@Nullable String value) {
+        if (value == null) {
+            return null;
+        }
+
+        final int index = value.indexOf(':');
+        if (index < 0) {
+            // no prefix
+            return value;
+        }
+
+        final String prefix = value.substring(0, index);
+        final String rest = value.substring(index + 1);
+
+        for (ConfigValueConverter converter : CONFIG_VALUE_CONVERTERS) {
+            if (converter.supportedPrefixes().contains(prefix)) {
+                return converter.convert(prefix, rest);
+            }
+        }
+
+        // return the value as is instead of rest because the value might contain ':' without prefix.
+        // e.g. "@#$:@#$_is_not_prefix"
+        return value;
```

Comment:
> What's the benefit of throwing an exception you think

Say that `asdf:<encrypted value>`  is used but the user forgets to specify an SPI.
For inputs that are not well defined, the encrypted value might be used as-is without the user being aware.

Reply:
Ah ha, that's a good suggestion. One of the reasons that I didn't throw an exception is that a user already contains `:` in their secret so that irrelevant exception is raised.
However, because there are not so many users who run Central Dogma, I think we can make a breaking change and provide the documentation about the prefix. 😉